### PR TITLE
Debug case report acmg #4172

### PR DESCRIPTION
**Pull request by dnil**
This PR adds a functionality or fixes a bug.

- Fix https://github.com/Clinical-Genomics/scout/issues/4172

<details>
<summary>Testing on cg-vm1 server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. Make sure the PR is pushed and available on [Docker Hub](https://hub.docker.com/repository/docker/clinicalgenomics/scout-server-stage)
1. Fist book your testing time using the Pax software available at [https://pax.scilifelab.se/](https://pax.scilifelab.se). The resource you are going to call dibs on is `scout-stage` and the server is `cg-vm1`.
1. `ssh <USER.NAME>@cg-vm1.scilifelab.se`
1. `sudo -iu hiseq.clinical`
1. `ssh localhost`
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `podman ps`
1. Stop the service with current deployed branch: `systemctl --user stop scout.target`
1. Start the scout service with the branch to test: `systemctl --user start scout@<this_branch>`
1. Make sure the branch is deployed: `systemctl --user status scout.target`
1. After testing is done, repeat procedure at [https://pax.scilifelab.se/](https://pax.scilifelab.se), which will release the allocated resource (`scout-stage`) to be used for testing by other users.
</details>

<details>
<summary>Testing on hasta server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. `ssh <USER.NAME>@hasta.scilifelab.se`
1. Book your testing time using the Pax software. `us; paxa -u <user> -s hasta -r scout-stage`. You can also use the WSGI Pax app available at [https://pax.scilifelab.se/](https://pax.scilifelab.se).
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `conda activate S_scout; pip freeze | grep scout-browser`
1. Deploy the branch to test: `bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_scout -t scout -b <this_branch>`
1. Make sure the branch is deployed: `us; scout --version`
1. After testing is done, repeat the `paxa` procedure, which will release the allocated resource (`scout-stage`) to be used for testing by other users.
</details>

**How to test**:
1. go to a cancer case, and pin a variant
2. note a crash when pulling general case report
3. apply patch
4. note general case report working

**Expected outcome**:
The functionality should be working
Take a screenshot and attach or copy/paste the output.

**Review:**
- [ ] code approved by
- [ ] tests executed by

### CODE REVIEW

File: scout/server/blueprints/cases/templates/cases/case_report.html [hunk 1]
```diff
@@ -1334,12 +1334,16 @@ <h6 class="card-subtitle mb-2 text-muted"><span class="badge badge-warning">Part
           </tr>
         </thead>
         <tbody>
+
         {% for acmg in ["P","LP", "VUS", "LB", "B", ] %}
-          {% for variant in variants|selectattr("acmg_classification.short", "equalto", acmg)|selectattr("first_rep_gene.hgnc_symbol", "defined")|sort(attribute='first_rep_gene.hgnc_symbol')%}
+          {% for variant in variants|selectattr("acmg_classification", "defined")|selectattr("acmg_classification.short", "equalto", acmg)|selectattr("first_rep_gene.hgnc_symbol", "defined")|sort(attribute='first_rep_gene.hgnc_symbol') %}
```
Comment (northwestwitch) [inline scout/server/blueprints/cases/templates/cases/case_report.html:1339]: This solution works, but you might want to trim the variant HGVS when it's too long. We had a dedicated PR not so long ago..

image

Comment (dnil) [reply scout/server/blueprints/cases/templates/cases/case_report.html:1339]: Good point; we should have like a `pretty_print for the report as well...

**Comment by dnil** [comment]
The also still doesn't explain why the underlying issue popped up now.

If you revert back to 4.71, everything still works, checkout `simplify_case_report` and it breaks. Now, that does not mean it is the only thing, only that it is sufficient to uncover the problem.

**Comment by northwestwitch** [comment]
> If you revert back to 4.71, everything still works,

Not for me: 

image

Sorry, I can't reproduce. again! 😆

File: scout/server/blueprints/cases/templates/cases/case_report.html [hunk 2]
```diff
@@ -1334,12 +1334,16 @@ <h6 class="card-subtitle mb-2 text-muted"><span class="badge badge-warning">Part
           </tr>
         </thead>
         <tbody>
+
         {% for acmg in ["P","LP", "VUS", "LB", "B", ] %}
```
Comment (northwestwitch) [inline scout/server/blueprints/cases/templates/cases/case_report.html:1338]:
```suggestion
        {% for acmg in ["P","LP", "VUS", "LB", "B" ] %}
```

**Comment by dnil** [comment]
> > If you revert back to 4.71, everything still works,
> 
> Not for me:
> 
> image
> Sorry, I can't reproduce. again! 😆

I assume you remembered pydantic?

**Comment by dnil** [comment]
> > > If you revert back to 4.71, everything still works,
> > 
> > 
> > Not for me:
> > image
> > Sorry, I can't reproduce. again! 😆

It's late, dont worry. It will reproduce like a charm tomorrow. Can I bring to mind that we had 4.71 running for what, a month? I sincerely doubt we had no cancer case reports with pinned variants generated during that time.

**Comment by dnil** [comment]
✅ Works on stage as well
Screenshot 2023-10-25 at 20 51 30

**Pull request merged**